### PR TITLE
IPA codebase synchronization

### PR DESCRIPTION
**Pull request by w0lek**
This PR contains IPA code synchronization with Aurora, and aggregates several tickets: feature, bugfix and code refactoring.

Changelog:
- IPA new way of parsing the VPR timing report based on attached report metadata. **This feature is requred to work together with the VPR IPA functionality, when it will be accepted in upstream.** 
- ReportModelLoader perform all heavy load in separate thread to not block the UI thread.
- Fix bug of slow critical path element selection with huge number of sub path elements (~10.000 elements produce delay about 8 seconds in past)
- IPA VIewer now has busy indicator while heavy work in process.
- Different code refactor enhancements.

### CODE REVIEW

**Comment by w0lek** [comment]
@ravikiranchollangi @volodymyrkochyn 
I put PR to draft mode, to block merging PR.
But PR is actually ready for review, i don't plan to make any other fixes, except some which you may ask me to correct after reviewing it.

**Comment by ravikiranchollangi** [comment]
> @ravikiranchollangi @volodymyrkochyn 
> 
> I put PR to draft mode, to block merging PR.
> 
> But PR is actually ready for review, i don't plan to make any other fixes, except some which you may ask me to correct after reviewing it.

For formatting use this:
.github/bin/run-clang-format.sh

There's also windows ci failure.

**Comment by w0lek** [comment]
@ravikiranchollangi @volodymyrkochyn
CI stages looks all passed

File: src/InteractivePathAnalysis/NCriticalPathWidget.cpp [hunk 1]
```diff
@@ -34,34 +34,30 @@
 #include <QPushButton>
 #include <QVBoxLayout>
 
+#ifndef TODO_IPA_MIGRATION_SETTINGS
+#include "../Compiler/QLSettingsManager.h"
```
Comment (volodymyrkochyn) [inline src/InteractivePathAnalysis/NCriticalPathWidget.cpp:38]: I don't see this file in the list. It looks like build will be broken with this flag.

Comment (volodymyrkochyn) [reply src/InteractivePathAnalysis/NCriticalPathWidget.cpp:38]: Closing this.

**Review by ravikiranchollangi — CHANGES_REQUESTED** [review]
Blocking this for logger discussion plus we have an ongoing release to add this now.

**Comment by ravikiranchollangi** [comment]
> Blocking this for logger discussion plus we have an ongoing release to add this now.

Need to move the IPA log file to timing analysis task.

**Comment by w0lek** [comment]
> > Blocking this for logger discussion plus we have an ongoing release to add this now.
> 
> Need to move the IPA log file to timing analysis task.

corrected here
https://github.com/os-fpga/FOEDAG/pull/1591/commits/7ca2411a95edbd9fb8575506303c349c7e3cfd4c

@volodymyrkochyn may you please review that change, is it ok?

**Comment by volodymyrkochyn** [comment]
@w0lek looks good to me

**Review by ravikiranchollangi — APPROVED** [review]

**Pull request merged**